### PR TITLE
smart flooder pipeline

### PR DESCRIPTION
**Pull request by kostekIV**

### CODE REVIEW

File: shell.py [hunk 1]
```diff
@@ -653,3 +654,33 @@ def setup_prometheus(region=default_region(), tag='prometheus', target_regions=u
...
+    call('yarn redspot compile'.split(), cwd='./bin/contracts-cli/deploy')
+    call('cargo clean'.split(), cwd='./bin/contracts-cli/deploy')
+    call('zip -r ./repo.zip ./contracts-cli'.split(),  cwd='./bin')
+
+    color_print('setup ...')
+    run_task('setup-contract-repo', regions=[region], tag=tag, pids=pids)
+
+
+def start_smart_flooder(signer='//Alice', methods_to_call=None, n_of_calls=None, region=default_region(), tag='dev', pids=None):
+    script = '#!/usr/bin/env bash\n' \
```
Comment (Marcin-Radecki) [inline shell.py:672]: Perhaps we can use `"""<multi_line_string> """` here?

File: shell.py [hunk 2]
```diff
@@ -653,3 +654,33 @@ def setup_prometheus(region=default_region(), tag='prometheus', target_regions=u
...
+        + f'cd contracts-cli\n' \
+        + 'export NVM_DIR="/home/ubuntu/.nvm"\n' \
+        + '[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh" \n' \
+        + 'export PATH="/home/ubuntu/.cargo/bin:${PATH}"\n' \
+        + 'export PATH="/home/ubuntu/binaryen-version_105/bin:${PATH}"\n' \
+        + 'nvm use\n' \
+        + 'rm -r deploy/node_modules flood/node_modules\n' \
+        + f'./deploy-and-flood.sh -s {signer} -n {n_of_calls or 100}  {" ".join(methods_to_call) if methods_to_call  else ""}'
+
+    with open('bin/smart_flood.sh', 'w') as f:
```
Comment (Marcin-Radecki) [inline shell.py:682]: I'm not familiar with this repo, is there any exception policy in this repo? I mean e.g. if this line throws, is it caught somewhere?

Comment (kostekIV) [reply shell.py:682]: nope, we usually use these functions from ipython console so all exceptions are just information for us what went wrong

Comment (kostekIV) [reply shell.py:682]: so basically they are meant to be run interactively,

Comment (Marcin-Radecki) [reply shell.py:682]: Asking since usually IO is thing that pops up exceptionally like lack of disk space, permissions, etc

File: shell.py (hunk 1 above)
Comment (kostekIV) [reply shell.py]: done

File: shell.py (hunk 2 above)
Comment (Marcin-Radecki) [reply shell.py:684]: Okay.

**Review by Marcin-Radecki — APPROVED** [review]
Changes looks good.

**Pull request merged**